src/routes/mod_creation.rs

```diff
@@ -0,0 +1,401 @@
+                        }
+                    } else if file_extension == "jar".to_string() {
+                        let initial_version_data = create_data
+                            .initial_versions
+                            .iter()
+                            .position(|x| x.file_indexes.contains(&current_file_index));
+
+                        if let Some(version_data_index) = initial_version_data {
+                            let version_data = create_data
+                                .initial_versions
+                                .get(version_data_index)
+                                .ok_or(CreateError::MissingValueError(
+                                    "Missing file extension!".to_string(),
+                                ))?
+                                .clone();
+
+                            let mut created_version_filter = created_versions
+                                .iter_mut()
+                                .filter(|x| x.number == version_data.version_slug);
+
+                            match created_version_filter.nth(0) {
+                                Some(created_version) => {
+                                    let upload_data = upload_file(
+                                        upload_url.get_ref().clone(),
+                                        "application/java-archive".to_string(),
+                                        format!(
+                                            "{}/{}/{}",
+                                            create_data.mod_namespace.replace(".", "/"),
+                                            version_data.version_slug,
+                                            file_name
+                                        ),
+                                        (&data).to_owned().to_vec(),
+                                    )
+                                    .await?;
+
+                                    created_version.files.push(VersionFile {
+                                        game_versions: version_data
+                                            .game_versions
+                                            .into_iter()
+                                            .map(|x| x.0)
+                                            .collect(),
+                                        hashes: vec![FileHash {
+                                            algorithm: "sha1".to_string(),
+                                            hash: upload_data.content_sha1,
+                                        }],
+                                        url: format!("{}/{}", cdn_url, upload_data.file_name),
+                                    });
+                                }
+                                None => {
+                                    let mut version_id = VersionId(random_base62(8));
+                                    //Check if ID is unique
+                                    loop {
```

Comment:
Same as above, infinite loop prevention - we should never be that close to full, but we want errors rather than stalling.